projects/dynamicforms/src/core/services/obo-customer/obo-customer.service.ts

```diff
@@ -0,0 +1,46 @@
+import { Injectable, OnDestroy } from '@angular/core';
+import { EventService, LogoutEvent, UserIdService } from '@spartacus/core';
+import { FSUser } from 'fsastorefrontlib/occ';
+import { BehaviorSubject, combineLatest, Observable, Subscription } from 'rxjs';
+import { map } from 'rxjs/operators';
+
+@Injectable({ providedIn: 'root' })
+export class OboCustomerService implements OnDestroy {
+  private subscription: Subscription = new Subscription();
+  private selectedCustomer: BehaviorSubject<FSUser> = new BehaviorSubject(null);
+
+  selectedCustomer$: Observable<FSUser> = this.selectedCustomer.asObservable();
+
+  constructor(
+    protected eventService: EventService,
+    protected userIdService: UserIdService
+  ) {
+    this.subscription.add(
+      this.eventService
+        .get(LogoutEvent)
+        .pipe(map(() => this.setSelectedCustomer(null)))
```

Comment:
Usually `tap` is a better operator to use in this case, since you are not transforming this stream.